### PR TITLE
Fix a lot of typings with strictNullChecks

### PR DESCRIPTION
**Pull request by azizhk**
https://medium.com/@martin_hotell/react-typescript-and-defaultprops-dilemma-ca7f81c661c7
https://github.com/Microsoft/TypeScript/pull/24422#issuecomment-395168742

https://app.bugsnag.com/anarock-dot-com/crm/errors/5bb85f475395b90019572036?event_id=5bb8b18100303d12acbe0000&i=sk&m=fq

### CODE REVIEW

**Comment by azizhk** [comment]
The test should not fail. Have changed only typings of that function, no code.

**Comment by azizhk** [comment]
[Typings](https://trello.com/c/TXdDuwU9/105-typings)

**Comment by azizhk** [comment]
@ritz078 So the failing test is actually a typescript error that it accepts a number but in the tests we are calling it with null.
How do you want to fix it? Accept only number and remove the null test? or keep the functionality as is and accept null as well.

**Comment by azizhk** [comment]
I've fixed by allowing null for now and added a few more tests.
Let me know and I'll make changes accordingly.

File: src/components/DropDown.tsx [hunk 1]
```diff
@@ -48,7 +48,7 @@ class DropDown extends React.PureComponent<DropdownProps, DropdownState> {
             labelComponent({ isOpen, toggleDropdown: this.toggleDropdown })
           ) : (
             <DropDownButton
-              isSelected={isSelected}
+              isSelected={!!isSelected}
```
Comment (ritz078) [inline src/components/DropDown.tsx]: why is this needed? isn't isSelected always boolean ?

**Comment by ritz078** [comment]
If in real code we don't want to accept null, we shouldn't accept that in test as well.

File: src/components/DropDown.tsx (hunk 1 above)
Comment (azizhk) [reply src/components/DropDown.tsx]: No its optional in DropdownProps, so can be undefined. 😢

Comment (ritz078) [reply src/components/DropDown.tsx]: ok
